### ISSUE TITLE
Private DNS Zone Limited to 15 Tags via api-version=2018-09-01. Via portal you can add upto the 50 tag limit. 

### ISSUE THREAD

**Issue by stevensoave** [Azure/azure-rest-api-specs#13600]
https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/privateDnsZones/{privateZoneName}?api-version=2018-09-01

When using programmatic tools/cli/terraform to apply tags to our Private DNS resource we get an error:
privatedns.PrivateZonesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="BadRequest" Message="The resource tags are invalid. A maximum of 15 tags are allowed with keys no longer than 512 and values no longer than 256 characters."

But the resource expects more than 15 tags via the portal. Key/Value are not the issue, it's the api that is limiting only 15 tags. Portal is supporting 50 tags on same resource. 

Example of someone else seeing this limit via terraform: https://www.gitmemory.com/issue/terraform-providers/terraform-provider-azurerm/11032/803544043

I've also see the limit if using the AzureCLI and this API. 

Please can you fix or give an update as to when it will be fixed? 

Thanks

**Comment by felickz** [comment]
Similarly - adding tags via terraform 2.9 and only 6 of my 9 tags  are added ... if i goto portal and try to add it shows success but if i refresh the UI -- tags not added! 

Length does not appear to be an issue as if make the value for one of my 6  tags shorter and then try to add my 7th it will not add!  Even changing existing tags takes some finessing...

**Comment by mblant** [comment]
Found this while working on a different issue.
New API version, 2021-04-01, allows for 50 tags 

Tags - Create Or Update At Scope 
https://docs.microsoft.com/en-us/rest/api/resources/tags/create-or-update-at-scope

PUT https://management.azure.com/subscriptions/{{subscriptionId}}/resourcegroups/{{resourcegroupName}}/providers/Microsoft.Network/privateDnsZones/{{zoneName}}/providers/Microsoft.Resources/tags/default?api-version=2021-04-01

Body:
{
    "properties": {
      "tags": {
        "tagKey1": "tag-value-1",
        "tagKey2": "tag-value-2",
        .....
        "tagKey50": "tag-value-50"
      }
    }
  }

**Comment by wvannuffele4** [comment]
> Found this while working on a different issue. New API version, 2021-04-01, allows for 50 tags

Did you actually test this? Does not work in my testing, neither does the newest 2022-09-01 version. 

`az rest --method PUT --uri "/subscriptions/{{subscription_id}}/resourceGroups/{{rg}}/providers/Microsoft.Network/privateDnsZones/{{dnszone}}/providers/Microsoft.Resources/tags/default?api-version=2021-04-01" --body "{\"properties\":{\"tags\":{\"a\":\"a\",\"b\":\"b\",\"c\":\"c\",\"d\":\"d\",\"e\":\"e\",\"f\":\"f\",\"g\":\"g\",\"h\":\"h\",\"i\":\"i\",\"j\":\"j\",\"k\":\"k\",\"l\":\"l\",\"m\":\"m\",\"n\":\"n\",\"o\":\"o\",\"p\":\"p\",\"q\":\"q\",\"r\":\"r\",\"s\":\"s\",\"t\":\"t\",\"u\":\"u\",\"v\":\"v\",\"w\":\"w\",\"x\":\"x\",\"y\":\"y\",\"z\":\"z\"}}}"`

-> 
`Bad Request({"error":{"code":"ProviderError","message":"{\"code\":\"BadRequest\",\"message\":\"The resource tags are invalid. A maximum of 15 tags are allowed with keys no longer than 512 and values no longer than 256 characters.\"}"}})`

`az rest --method PUT --uri "/subscriptions/{{subscription_id}}/resourceGroups/{{rg}}/providers/Microsoft.Network/privateDnsZones/{{dnszone}}/providers/Microsoft.Resources/tags/default?api-version=2022-09-01" --body "{\"properties\":{\"tags\":{\"a\":\"a\",\"b\":\"b\",\"c\":\"c\",\"d\":\"d\",\"e\":\"e\",\"f\":\"f\",\"g\":\"g\",\"h\":\"h\",\"i\":\"i\",\"j\":\"j\",\"k\":\"k\",\"l\":\"l\",\"m\":\"m\",\"n\":\"n\",\"o\":\"o\",\"p\":\"p\",\"q\":\"q\",\"r\":\"r\",\"s\":\"s\",\"t\":\"t\",\"u\":\"u\",\"v\":\"v\",\"w\":\"w\",\"x\":\"x\",\"y\":\"y\",\"z\":\"z\"}}}"`

->
`Bad Request({"error":{"code":"ProviderError","message":"{\"code\":\"BadRequest\",\"message\":\"The resource tags are invalid. A maximum of 15 tags are allowed with keys no longer than 512 and values no longer than 256 characters.\"}"}})`


The exact same call targeting other types of resources, such as a storage account, works perfectly.

**Comment by sghodke-wowcorp** [comment]
Hi Azure team,

  Please suggest about error while applying more than 15 tags on Private DNS zone.
  
 I am also facing this error.

**Comment by eehret** [comment]
Just ran into this one myself and was quite surprised to find the (seemingly) arbitrary limit?

**Comment by eehret** [comment]
https://github.com/hashicorp/go-azure-sdk/blob/main/resource-manager/privatedns/2020-06-01/virtualnetworklinks/version.go

I guess that needs updating?

**Comment by andywilde** [comment]
This caused our AKS clusters to go into a failed state. By simply adding one extra tag (totalling 16) on the AKS service, which is then inherited by the Private DNS zone, the cluster was no longer able to autoscale.

**Comment by ausfestivus** [comment]
> This caused our AKS clusters to go into a failed state

Thats worthy of a support case and please-explain to the AM. Bonus points because you get to ask them why an issue that has been known about for 3 years brought down prod.

**Comment by levimm** [comment]
> This caused our AKS clusters to go into a failed state. By simply adding one extra tag (totalling 16) on the AKS service, which is then inherited by the Private DNS zone, the cluster was no longer able to autoscale.

@andywilde We'll update the api version used in AKS and support 50 tags in next release. ETA for the rollout is mid May

**Comment by andywilde** [comment]
> > This caused our AKS clusters to go into a failed state
> 
> Thats worthy of a support case and please-explain to the AM. Bonus points because you get to ask them why an issue that has been known about for 3 years brought down prod.

Yes, we did raise a support request but were told "There is a disclaimer that for some resources, 15 tags is the limit, which includes Azure DNS Zone. As we saw, the error log it explicitly inform the error is due to a maximum of 15 tags, and does not matter from where the request is coming from (Azure Portal, CLI, Terraform) or the API version, it just about the limit of the resource." We know that is not true since Private DNS Zone *can* be updated to more than 15 tags from the Azure Portal.

**Comment by levimm** [comment]
@andywilde  strange since I just did a test myself and both portal and all api-version with cli doesn't work with more than 15 tags.

**Comment by andywilde** [comment]
@levimm I just checked on the Azure portal and I can't set more than 15 tags either. Looks like I assumed this bug was still valid without checking.  Can anyone actually still do this on Azure Portal?

**Comment by Manbearpiet** [comment]
This is really annoying, we're running into this as well. Same for API-Management and Azure Automation Accounts..

**Comment by porrascarlos802018** [comment]
All, consider submitting your feedback and vote here: https://feedback.azure.com/d365community/idea/4d4bd32f-dcb9-ed11-a81b-6045bd79fc6e

**Comment by vandriot** [comment]
Hi, 

I just encountered the issue as well. 
Any news from Azure team on this issue please? @raych1 @aznetsuppgithub

Thanks.

**Comment by raych1** [comment]
@aznetsuppgithub could you provide feedback on this issue or route to the appropriate contacts?

**Comment by vandriot** [comment]
Hi @raych1,

Thank you for your feedback. 
I opened a case with Azure support regarding this issue and got the following feedback:

_We understand that while trying to add more than 15 tags to a private DNS zone, you are getting an error that states you are reaching a limit.
Unfortunately, this limitation exists for private and public zones by design, and has to do with how these tags are stored in our backend for these services.
Our engineering is aware this limitation impacts some scenarios and has taken it into account in their discussions.
Currently, we have found no plans to change this behavior.
In the future, our engineering may consider changing this behavior if a large enough user base requests this feature and justifies the effort required._

So it is a known limitation for this service, until a fix is deployed by Azure engineering teams, there is not a lot more we can do here unfortunately.

**Comment by jwestfall-MSFT** [comment]
If tags are stored in the backend service in such a way that it's limited to 15, why does this limitation not exist when creating through the portal?

**Comment by vladmbjss** [comment]
" if a large enough user base requests this feature" - how large does this have to be? For us corporate Policy is enforcing 15 FinOps related tags, leaving us unable to add project related tags to DNS zone.

**Comment by jaredbrogan** [comment]
This is really annoying. Would be really cool if Azure would actually hold all of their APIs to the same standard, because honestly little issues like this seem to happen all of the time due to the fact that there seemingly is zero consistency across the board. Do better.
